### ISSUE TITLE
Max number of items per row

### ISSUE THREAD

**Issue by aytunch** [hauketoenjes/responsive_grid_list#3]
When my app works in web, the width is very big as expected. But I don't want to see 10 grid items next to each other.

Is there a way to limit the maximum number of grid items per row?

Thanks

**Comment by hauketoenjes** [comment]
Hi, thanks for the feature request 👍🏼 .

I'll add this feature in the next release in a couple of days.

**Comment by hauketoenjes** [comment]
I added the `maxItemsPerRow` option in all available Widgets and the `AbstractResponsiveGridList`.

[Version 1.2.3](https://pub.dev/packages/responsive_grid_list/versions/1.2.3) is available on pub.dev

**Comment by aytunch** [comment]
Thank you very much.
I tried it and it works. However when i set `maxItemsPerRow` to 3 and increase the web pages width, the items get huge.
Just like `minItemWidth`, can we have a `maxItemWidth` to limit item widths? So that there is space around each row.

**Comment by hauketoenjes** [comment]
Yeah, I also noticed that behavior. It is probably a good idea to add `maxItemWidth` and just fill the remaining space around. I'll look into this the next days. 

Maybe it is a good idea to also add options to control where the space is added.  Either between the items, or at the edges.

**Comment by aytunch** [comment]
Yeah, giving an option to choose between spreading the extra space either to edges or between items would be great.
For my specific case(web), the edges make more sense.

Take care.

**Comment by hauketoenjes** [comment]
I recently stumbled across the same thing in a web project of mine and thought a bit about this issue. 

I am not convinced that this is an issue this package should address. Personally, I solved it by just constraining the parent widget of the Grid to a max width (with a `ConstrainedBox` for example). That has the same effect that you are describing (spacing on the sides).

I think this should be the preferred way of doing it.
